src/account_manager/builder.rs

```
@@ -134,90 +134,89 @@ impl AccountManagerBuilder {
-                        None => data
-                            .client_options
-                            .ok_or(crate::Error::MissingParameter("client_options"))?,
-                    };
-                    let secret_manager = match self.secret_manager {
-                        Some(secret_manager) => secret_manager,
-                        None => data
-                            .secret_manager
-                            .ok_or(crate::Error::MissingParameter("secret_manager"))?,
-                    };
-                    let coin_type = match self.coin_type {
-                        Some(coin_type) => coin_type,
-                        None => data
-                            .coin_type
-                            .ok_or(crate::Error::MissingParameter("coin_type (IOTA: 4218, Shimmer: 4219)"))?,
-                    };
-                    (client_options, secret_manager, coin_type)
-                }
-                // If no account manager data exist, we will set it
-                None => {
-                    // Store account manager data in storage
-                    storage_manager.lock().await.save_account_manager_data(&self).await?;
-                    (
-                        self.client_options
-                            .ok_or(crate::Error::MissingParameter("client_options"))?,
-                        self.secret_manager
-                            .ok_or(crate::Error::MissingParameter("secret_manager"))?,
-                        self.coin_type
-                            .ok_or(crate::Error::MissingParameter("coin_type (IOTA: 4218, Shimmer: 4219)"))?,
-                    )
-                }
-            };
-
-            let client = client_options.clone().finish()?;
-
-            let accounts = storage_manager.lock().await.get_accounts().await.unwrap_or_default();
+        let read_manager_builder = storage_manager.lock().await.get_account_manager_data().await.ok();
```

Not a big fan of this `.ok()`, it's silencing errors